### ISSUE TITLE
can't save subtitles

### ISSUE THREAD

**Issue by damianatorrpm** [flathub/org.videolan.VLC#70]
works with non-flatpak vlc
doesnt work with flatpak vlc
vlc is not allowed to save the subtitle file with flatpak

**Comment by MatMaul** [comment]
@damianatorrpm After a quick investigation it seems to be a bug when the folder where it stores the subs is not already created.
Can you try to do `mkdir ~/.var/app/org.videolan.VLC/cache/vlc` ? It fixes the problem for me. I'll work on a real fix for VLC.

**Comment by damianatorrpm** [comment]
@MatMaul Thank you for the response. I don't know why but it is fixed in fedora rawhide with latest VLC flatpak for me anyways.

**Comment by MatMaul** [comment]
@damianatorrpm can you try to do `rm ~/.var/app/org.videolan.VLC/cache/vlc` and report the result ? It should break it again.
This folder was probably created by the use of another functionality of VLC using it, hence the fact that it works for you now.
Just do a mkdir to get back to a working state ;)
Thanks.

**Comment by damianatorrpm** [comment]
@MatMaul Yes it breaks it again! recreating the folder, restarting vlc, and trying again fixes the issue ones again

**Comment by nazar2sfive** [comment]
`mkdir ~/.var/app/org.videolan.VLC/cache/vlc`

**it solved for me in clear linux.**

**Comment by damianatorrpm** [comment]
Recently the following problem started:
Error: code '408 invalid parameters; moviebytesize is not number' (200)

**Comment by LargePrime** [comment]
> `mkdir ~/.var/app/org.videolan.VLC/cache/vlc`
> 
> **it solved for me in clear linux.**

thanks for the fix

**Comment by ItaloQualisoni** [comment]
Thanks! This has solved my issue too for flatpak VLC in Fedora 32 

> `mkdir ~/.var/app/org.videolan.VLC/cache/vlc`
> 
> **it solved for me in clear linux.**

**Comment by pacho2** [comment]
Supposedly it should create the needed paths... but it fails. 

```
  openSub.conf.dirPath = vlc.config.userdatadir()
  local subdirs = { "lua", "extensions", "userdata", "vlsub" }
  for _, dir in ipairs(subdirs) do
    local res, err = vlc.io.mkdir( openSub.conf.dirPath .. slash .. dir, "0700" )
    if res ~= 0 and err ~= vlc.errno.EEXIST then
      vlc.msg.warn("Failed to create " .. openSub.conf.dirPath .. slash .. dir )
      return false
    end
    openSub.conf.dirPath = openSub.conf.dirPath .. slash .. dir
  end
```

That is from share/lua/extensions/VLSub.lua from vlc-3.0.10 sources
